crates/walrus-service/src/node/blob_event_processor.rs

```diff
@@ -203,21 +203,43 @@ impl BackgroundEventProcessor {
 
         let histogram_set = self.node.metrics.recover_blob_duration_seconds.clone();
 
-        if !self.node.is_blob_certified(&event.blob_id)?
-            // For blob extension events, the original blob certified event should already recover
-            // the entire blob, and we can skip the recovery.
-            || event.is_extension
+        // Get the current event epoch to check if the blob is fully stored up to the shard
+        // assignment in this epoch. If the current event epoch is not set (the node may just
+        // started), we always start the blob sync.
+        let current_event_epoch = self.node.try_get_current_event_epoch();
```

Comment:
Question: (Why) can't we just wait for this, i.e., call `self.node.current_event_epoch().await`?

Reply:
The problem is following, say after restart, first event is blob extension event, and the second event is epoch change start event. Blob extension event won't set the `current_event_epoch()`, therefore this will block. However, before processing epoch change start event, we need to [wait until all prior events are processed](https://github.com/MystenLabs/walrus/blob/27ef5a3922e5206f160e7ffd41b2fe22e1486c05/crates/walrus-service/src/node.rs#L1554). Because of this, there is a deadlock.

I've thought for quite sometime how to address this, and by initiating the blob sync is the best simple solution I can come up right now.

Reply:
Ah, this is a very good point, I didn't think about this case. 😬 
Good that you thought about it!